sdk/search/Azure.Search/src/Models/IndexDocumentsAction{T}.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using Azure.Core;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+namespace Azure.Search.Models
+{
+    // Hide the untyped IndexAction
+    [CodeGenSchema("IndexAction")]
+    internal partial class IndexAction { }
+
+    /// <summary>
+    /// Represents an index action that operates on a document.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The .NET type that maps to the index schema. Instances of this type can
+    /// be retrieved as documents from the index.
+    /// </typeparam>
+    public class IndexDocumentsAction<T> : IUtf8JsonSerializable
+    {
+        /// <summary>
+        /// The operation to perform on a document in an indexing batch.
+        /// </summary>
+        public IndexActionType ActionType { get; private set; }
+
+        /// <summary>
+        /// The document to index.
+        /// </summary>
+        public T Document { get; private set; }
+
+        /// <summary>
+        /// Private constructor to prevent direct instantiation.
+        /// </summary>
+        private IndexDocumentsAction() { }
+
+        /// <summary>
+        /// Initializes a new instance of the IndexDocumentsAction class.
+        /// </summary>
+        /// <param name="type">
+        /// The operation to perform on the document.
+        /// </param>
+        /// <param name="doc">The document to index.</param>
+        public IndexDocumentsAction(IndexActionType type, T doc)
+        {
+            Debug.Assert(Enum.IsDefined(typeof(IndexActionType), type));
```

Comment:
Shouldn't you be using the `Argument` class to assert at leat that `doc` isn't null? In what cases does it make sense for it to be null here?

Reply:
I remember why I didn't do this.  We don't have a class constraint on `T` because we allow structs.